src/backend/distributed/README.md

```diff
@@ -0,0 +1,2079 @@
+# Citus Technical Documentation 
+
+The purpose of this document is to provide comprehensive technical documentation for Citus, in particular the distributed database implementation. 
+
+# Citus Concepts 
+
+**Citus table**: A table managed by Citus through PostgreSQL hooks. Whenever the table is involved in a command, the command is handled by Citus. 
+
+**Shell table**: The “original” (inaccessible) PostgreSQL table which has been converted to a Citus table. Every node has shell tables and the corresponding Citus metadata. 
+
+**Metadata**: The Citus catalog tables, shell tables, and dependent objects that are replicated to all nodes to enable distributed queries from those nodes. 
+
+**Metadata syncing**: The process of replicating metadata from the coordinator to other nodes, which happens when adding a node or any change to the metadata. 
+
+There are several types of Citus tables: 
+
+**Distributed tables** are created using SELECT create_distributed_table(..). They have a distribution column and for each row the value in the distribution column determines which to shard the row is assigned. There are 3 different partitioning schemes for distributed tables, though only the first is supported: 
+
+- Hash-distributed tables have a range of hash values in shardminvalue/shardmaxvalue in pg_dist_shard 
+- Range-distributed tables (deprecated) have shards with a distinct range of values in pg_dist_shard 
+- Append-distributed tables (deprecated) have shards with a range of values in pg_dist_shard, though the ranges can overlap. 
+
+Hash-distributed tables can be **co-located** with each other, such that the shards with the same hash value range are always on the same node. From the planner point-of-view, range-distributed tables can also be colocated, but shards are not managed by Citus.  
+
+Shards can be replicated (deprecated), in which case they have multiple shard placements. Writes to the placements are performed using 2PC and use aggressive locking to avoid diverging. 
+
+**Reference tables** are created using SELECT create_reference_table(..). They have a single shard, which is replicated to all nodes. It is possible to have a node without reference table replicas, but in that case the reference tables are replicated before the next rebalance. Reference tables are always co-located with each other and have a common co-location ID. 
+
+Writes to a reference table are performed using 2PC and use aggressive locking to avoid diverging. 
+
+**Single shard tables** are a special type of distributed table without a distribution column and with a single shard. When using schema-based sharding, tables created in a distributed schema automatically become single shard tables. Single shard tables can be co-located with each other, but not replicated. 
```

Comment:
For distributed and reference tables you name the UDF that creates them. What aout for single-shard tables and Citus local tables?

Reply:
makes sense